### ISSUE TITLE
Dockerfile

### ISSUE THREAD

**Issue by scientes** [HackerSpace-PESU/Best11-Fantasycricket#39]
How do you plan on hosting the App?

**Describe the solution you'd like**
Using Docker/docker-compose would one of the easier ways of hosting, especially it you plan on using a database. It would also make using a database in development more comparable when issues arise

**Comment by lucasace** [comment]
Seems interesting , Was actually planning to learn Docker , might as well implement it
But I dont want to host it until the website is dynamic i.e it uses daily match schedules instead of the test cases
But we can be keep it ready though
Thank you for the suggestion
Also If you were planning to start this. I would recommend waiting till 18th of Nov since I'm sure you noticed the clone time is way too long and takes 20 MB . We are planning to reduce that which might mess the commit history. If you start now, syncing might not work

**Comment by scientes** [comment]
jup thats because of the large data files

**Comment by scientes** [comment]
lfs might be a solution: https://git-lfs.github.com/

**Comment by lucasace** [comment]
The data files arent that huge. You can check it if you want. Its probably the images which were commited and then deleted . This got stored in the pack folder of .git. Thus while cloning it takes 20mb and recieves 5000 objects. We plan to change the images to links, so it can be used efficiently
We really didnt organize the repo during the start of the project xD

**Comment by scientes** [comment]
Lfs could solve the image Problem but links is probably easier

**Comment by scientes** [comment]
Also when looking at docker fastapi Has a good Image you can build on

**Comment by lucasace** [comment]
> Also when looking at docker fastapi Has a good Image you can build on

Sounds Interesting. Ill surely check it out soon enough once the cloning issue is solved

**Comment by scientes** [comment]
would be a minimal example (the filepaths are not modified to fit your project, it expects the main.py to be in /app)
```dockerfile
FROM tiangolo/uvicorn-gunicorn-fastapi:python3.8-slim

WORKDIR /app/
RUN pip install --no-cache-dir poetry; rm -r /tmp ; rm -r /root/.cache"


COPY ./app /app
ENV PYTHONPATH=/app
```

**Comment by lucasace** [comment]
> would be a minimal example (the filepaths are not modified to fit your project, it expects the main.py to be in /app)

Thank you, will get started soon 😄

**Comment by lucasace** [comment]
Closed by #59 thanks to @scientes
